tests/rptest/tests/rpk_group_test.py

```diff
@@ -67,13 +70,24 @@ def test_group_list_and_delete(self):
         self.rpk.consume(topic, group=group_2, n=10)
 
         # groups will show up until they're deleted or until all offsets expire
-        groups = self.rpk.group_list_names()
-        assert group_1 in groups
-        assert group_2 in groups
+        group_names = self.rpk.group_list_names()
+        assert group_1 in group_names
+        assert group_2 in group_names
 
         self.rpk.group_delete(group_1)
-        groups = self.rpk.group_list_names()
-        assert group_1 not in groups
+        group_names = self.rpk.group_list_names()
+        assert group_1 not in group_names
+
+        all_groups = self.rpk.group_list()
+        self._assert_eq(all_groups[0].group, group_2)
+        self._assert_eq(all_groups[0].state, "Empty")
+
+        e_s_groups = self.rpk.group_list(states=["Empty", "Stable"])
+        self._assert_eq(e_s_groups[0].group, group_2)
+        self._assert_eq(e_s_groups[0].state, "Empty")
```

Comment:
These two requests have the same outcome. Would it be worth adding a test for two requests that produce nonempty but different responses? No worries if it's a pain.

Reply:
I was thinking about this but it's not easy. I think we could test this with a non-empty response (the same way I was manually testing this locally) by starting a consumer in a separate thread and calling `group_list` in parallel. I think we already have reasonable coverage of state filtering with these tests and a multi-threaded test for this endpoint seems too much.

Reply:
yeah, i agree. a threaded test would be overkill